lib/basicauth.php

```diff
@@ -24,6 +24,31 @@ public function __construct($authUrl) {
 	 * @return true/false
 	 */
 	public function checkPassword($uid, $password) {
+		/*
+		 * Connect without user/name password to make sure
+		 * URL is indeed authenticating or not...
+		 */
+		stream_context_set_default(array(
+		  'http'=>array(
+		    'method'=>"GET",
+		  ))
+		);
+		$headers = get_headers($this->authUrl, 1);
```

Comment:
could you use an other variable name here for the headers for this check? (so we can make sure this is never mixed up with the real auth request somehow)

Reply:
Renamed the variable to $canary.